### ISSUE TITLE
NBbomber Cloud

### ISSUE THREAD

**Issue by arkiaman** [VIP-Logic/NBomber#650]
Hi,

Do you have any roadmap for NBomber Cloud offer availability ?

**Comment by AntyaDev** [comment]
Hi @arkiaconsulting 

We have a plan to release a first beta in the middle of Spring (in worst case end of Spring).
But thanks for the question, I will update our roadmap regarding NBomber Cloud.

**Comment by arkiaman** [comment]
If you ever search for some guinea pigs, I would be pleased.

**Comment by arkiaman** [comment]
Hi @AntyaDev ! Is this feature still on your roadmap ?

**Comment by AntyaDev** [comment]
Hi @arkiaconsulting ,
Yes, it's in active development. We already have a prototype that allows you to schedule jobs that will run the NBomber script in the cloud. Right now, we are polishing the system and adding basic management features: filtering, editing jobs, etc.

**Comment by BKuatche** [comment]
Hello @AntyaDev ,
I was wondering if there’s any update on NBomber Cloud. We’re currently planning to use Azure Load Testing with NBomber, and I’d like to check if there are any new developments or guidance around this integration.

**Comment by AntyaDev** [comment]
Hi @BKuatche 
Yes, we have made some progress and will prepare some updates regarding this.
Could you please tell me: do you use Kubernetes in your organisation?

**Comment by BKuatche** [comment]
Hello @AntyaDev 
Thank you for your quick response.
We currently utilize Kubernetes through AKS for our workloads. I’m also curious to understand how you are planning to incorporate SignalR into the load testing approach, as that’s an area of particular interest for us.

**Comment by AntyaDev** [comment]
@BKuatche That's cool!
We’ve been working on [NBomber Studio](https://nbomber.com/docs/nbomber-studio/overview), which allows you to create a load test and deploy it to Kubernetes (K8s).
We call this feature **Private Cloud**. Essentially, it takes care of running and deploying your NBomber distributed (cluster) test in K8s for you. In addition, you’ll be able to view real-time reporting directly in NBomber Studio. We are currently testing it and plan to release it in late November or December.

I suggest you subscribe to our [LinkedIn ](https://pl.linkedin.com/company/nbomber) to receive such updates.

Regarding SignalR — it’s great that you decided to cover those workflows with NBomber. We’d be happy to help you with that. Could you describe your use case with SignalR? If you prefer privacy, you can email us or reach out via Slack.

**Comment by BKuatche** [comment]
Hello @AntyaDev 
Thank you for the update, it is much appreciated.

Regarding SignalR, we have clients that not only make multiple REST API calls but also connect to SignalR. Our objective is to simulate the load that the open SignalR connection and event mechanism puts on the environment, while at the same time generating load on the other REST APIs.

**Comment by AntyaDev** [comment]
@BKuatche Yes, your SignalR use case can be similar to any scenario involving long-lived connections. Essentially, you’ll need to create a SignalR client for each user and start listening for updates. In terms of semantics, it should be similar to working with [WebSockets](https://nbomber.com/docs/protocols/websockets). Also, you may need to consider using [ClientPool](https://nbomber.com/docs/nbomber/client-pool).

I suggest checking the [examples project](https://github.com/PragmaticFlow/NBomber/tree/dev/examples/Demo/WebSockets).
